### PR TITLE
Revising default list of comets

### LINKED ISSUE #561: Can't find 46P/Wirtanen
### Actual Behaviour
Describe or maybe attach a screenshot?

### System
* Stellarium version: <Name of downloaded installable file?>
* Operating system: <Name, version number>
* Graphics Card: <Manufacturer (likely Intel, NVidia, AMD?), Model (HD, Geforce, Radeon..., with model number), driver version?>

### Logfile 
If possible, attach the logfile `log.txt` from your user data directory. Look into the Guide for its location.

### LINKED ISSUE #1241: Incorrect 1P/Halley position near 1986 perihelion
### Expected Behaviour
Correct prediction of the position of Comet Halley at or near 1986 perihelion.
### Actual Behaviour

Issue #1: The orbital elements for comet 1P/Halley presently used as a default are seriously defective when asked to reproduce Halley's last appearance in 1986. Among other differences, the perihelion date of the real comet was Feb 9, 1986 [see Yeomans and Chodas Astron Jour v98 n3 p 1083 (Sept 89)]. The Stellarium default elements of 1P/Halley have the perihelion date of 21 Jan 1986! Stellarium shows the position of the comet on 18 Mar 1986 in Scorpius, and I know this is incorrect because I photographed it in Sagittarius. I recently downloaded a set of elements from the Minor Planet Center into Stellarium, and the revised elements are as follows"

Table: Epoch 1986 MPC revised elements matching contemporary photos, adapted to Stellarium format.

[1phalley]
orbit_good = 1000
orbit_Eccentricity = 0.967277
orbit_TimeAtPericenter = 2446470.958888889
name = 1P/Halley
orbit_ArgOfPericenter = 111.8657
albedo = 0.1
type = comet
absolute_magnitude = 4
orbit_Inclination = 162.2422
dust_brightnessfactor = 1.5
dust_lengthfactor = 0.4
slope_parameter = 6
orbit_visualization_period = 27758.20106745674
orbit_PericenterDistance = 0.587104
dust_widthfactor = 1.5
orbit_AscendingNode = 58.8601
coord_func = comet_orbit
radius = 5

The particular page where Stellarium obtained this information is, I believe,
https://www.minorplanetcenter.net/iau/MPEph/MPEph.html

Again, perihelion is Feb 9 in the elements of the Table. These revised elements (orbit epoch = Feb 19.0, 1986) were tested against contemporary photos and good matches were found (my photo on 18 Mar 1986 was less than an arcminute off). The results of downloading elements from several files distributed by the same MPC and available elsewhere in Stellarium were erroneous. They show an 18 Mar 1986 position between Sagittarius and Scorpius. 

The reason that the files being distributed by the same MPC use different elements from what it delivered in the ephemeris calculation service in the link above is unknown to me. Either the files are corrupted by a typo, or they refer to distant orbit epochs having little to do with the comet when it was visible. I can reproduce the JPL Small Body Database Browser (Epoch 17 Feb 1994) elements, but I have not been able to reproduce a perihelion date 18 days early.

Since it makes no sense to use orbital elements which place Halley when it was visible in the next constellation, I suggest the default elements be replaced by those in the Table.

Related Issue #2) Orbit_Epoch is not recorded with a comet's elements. Bodies should not be displayed outside of a window that is some sort of multiplier of orbit_good on either side of orbit_epoch, but I have been unable to find a parameter that would contain this information. Are all comets in Stellarium assumed to have orbit_epoch at perihelion? If so, that would explain a lot.

I have both of these issues discussed in more detail in my on-line essay
https://www.bay-astronomers.org/Backissues/CometHalley.pdf

### Steps to reproduce
Set the time to 1986 Mar 18. Witness the comet placed in Scorpius. The actual position is given in the following photograph. As a guide to the eye, the handle of the Teapot of Sagittarius is denoted by white lines, as is a little kite-shaped figure below the comet.  Please note that the error also exists at the stellarium-web.org site.

image

### System
* Stellarium version: <0.20.2 Windows>
* Operating system: <Windows 10 Home>
* Graphics Card: <Intel HD Graphics, 9.17.10.4459>
* Screen type (if applicable): Not Relevant -- No malfunction

### Logfile 
If possible, attach the logfile `log.txt` from your user data directory. Look into the Guide for its location.

### LINKED ISSUE #3261: Default list of comets
When I worked for partially fix an issue #3224 (see PR #3230) I got an important idea - we should revised the default list of comets: 
* The number of comets should be reduced (probably 10-15 comets at well - most bright and/or most famous and maybe expected be bright in near future)
* Historical data should contains correct designation and correct orbital elements (for observable epoch!) - to avoid mix with modern data the name should contains an asterix.
* The modern comets probably should be revised per every release

### PR DESCRIPTION
**Pull request by alex-w**
### Description
The default list of comets contains now a few very bright comets - at least visual magnitude 0 - so, this is good naked eye observed comets. 

Fixes #3261
Fixes #1241
Fixes #561

### Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] This change requires a documentation update

## Checklist:

- [x] My code follows the [code style](http://stellarium.org/doc/head/codingStyle.html) of this project.
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (header file)
- [ ] I have updated the respective chapter in the Stellarium User Guide
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by alex-w** [comment]
OK, the PR is ready to review & merge

File: plugins/SolarSystemEditor/resources/comet_discovery.fab [hunk 1]
```diff
@@ -89,7 +89,7 @@
    C/1583 J1|C/1582    J1|1582       |      |1582-05-12|T. Brahe
    C/1585 T1|C/1585    T1|1585       |      |1585-10-13|T. Brahe
    C/1590 E1|C/1590    E1|1590       |      |1590-03-05|T. Brahe
-   C/1593 O1|C/1593    O1|1593       |      |1593-07-29|Ch.J. Ripensis
```
Comment (gzotti) [inline plugins/SolarSystemEditor/resources/comet_discovery.fab:92]: The 2-letter abbreviations used in 16th-20th centuries were a greater help to almost always have the name right. 

Ph=Philipp vs. P=Paul/Peter/Pierre
Ch=Christ(oph(er) | ian)  vs. C=Carl (But French Ch=Charles, Chr=Christophe)

I would prefer the 2-letter style.

**Comment by gzotti** [comment]
Other than that, the list should be good for now (23.2). We could think about packing only 2020 and later in the default distribution, and add extra files to load for e.g. `showpieces 1950-2020`, `all comets 1950-2020`, and then shorten the 1000comet list to end in 1950 (to avoid overlap). But it's just tedious reformatting. If there is a better list to replace the 1000comets, fine for me. But for some of the historical comets I tried to match the tail parameters to photos or old prints, so these data should not get lost.

File: plugins/SolarSystemEditor/resources/comet_discovery.fab (hunk 1 above)
Comment (alex-w) [reply plugins/SolarSystemEditor/resources/comet_discovery.fab]: Done

**Comment by alex-w** [comment]
> Other than that, the list should be good for now (23.2). We could think about packing only 2020 and later in the default distribution, and add extra files to load for e.g. `showpieces 1950-2020`, `all comets 1950-2020`, and then shorten the 1000comet list to end in 1950 (to avoid overlap). But it's just tedious reformatting. If there is a better list to replace the 1000comets, fine for me. But for some of the historical comets I tried to match the tail parameters to photos or old prints, so these data should not get lost.

Probably it will be few loadable lists for old comets (`ancient`, `renaissance`, `middle age` or something other base for sorting the data), but I fear we should add some GUI elements in Solar System Editor plugin to manage the lists.  And what about non-cometary data in `ssystem_minor.ini` file? We need to split the data onto few files/lists or add something to correct manage whole SSO. 

The packing only 2020 and later in the default distribution may be acceptable, but what about periodical comets?

In any way, proposed task touches issue #3240, and it should be solving in separate PR, because it related to changing the architectures of SolarSystem class and SSE plugin.

**Comment by gzotti** [comment]
You can of course add some Epoch/Perihel date filtering to the "load data from file", but it's not topic of this PR and probably something for 23.3 or later.  If such a filter becomes reality, there is no need to split data files per antiquity/medieval/renaissance/.... The data files can be named by their original sources, the filtering will select what user wants. I assume this is a very small, specialized user group here. If there is a list "best comets since 1935", this should make even our oldest observer/users happy who may want to recreate their earliest observations from the 1950s. 

Periodical comets still need at least one entry per apparition, so it's no difference. (You cannot use data from some 2018 apparition to compute the 2024 or 2035 orbit.)

Non-comet minor objects: Not sure what "typical" users want. A few asteroids (1-20?), the dwarf planets, some KBOs should be there permanently, with updated elements per release. Smaller ones if they are something worth demonstrating. (Spacecraft visits/impacts, ...) There was one memorable event at the evening of the Chelyabinsk meteor fall, when a small object passed Earth. However, in these hours, its orbit was changed by Earth's gravity, so this cannot even be properly demonstrated with one element set (I had used elements from JPL/Horizon to create a chain of objects with elements 1 hour apart, but average users would be confused.).  Experienced observers will create their own set.
I would like to see the asteroid belt symbolized by a particle system driven by real elements for 5.000~250.000 asteroids computed on the GPU (Kepler elements, fixed 5 iterations). Identification of single dots will not be possible. This has been a wishlist topic for a plugin since Launchpad times.

**Comment by alex-w** [comment]
So, is the current list (for v23.2) good for you?

**Review by gzotti — APPROVED** [review]

**Comment by alex-w** [comment]
Oops... I found issue in the filters for comets/asteroids in the plugin and it fixed now. Plus I've added all comets, where discovered/recovered in years 2022-2023.

**Pull request merged**